tests/deepsparse/v2/middleware/test_middleware_manager.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from deepsparse.v2.pipeline import Pipeline
+from deepsparse.v2.routers import LinearRouter
+from deepsparse.v2.schedulers import ContinuousBatchingScheduler, OperatorScheduler
+from tests.deepsparse.v2.middleware.utils import CounterMiddleware, OpsTrackerMiddleware
+from tests.deepsparse.v2.test_basic_pipeline import (
+    AddOneOperator,
+    AddTwoOperator,
+    IntSchema,
+)
+
+
+def test_middleware_triggered_in_expected_order_for_ops():
+    """Check that when ops gets triggered, middleware behaves as expected"""
+    AddThreePipeline = Pipeline(
+        ops=[AddOneOperator(), AddTwoOperator()],
+        router=LinearRouter(end_route=2),
+        schedulers=[OperatorScheduler()],
+        continuous_batching_scheduler=ContinuousBatchingScheduler,
+        middleware=[OpsTrackerMiddleware(), CounterMiddleware()],
```

Comment:
let's define these two middleware as objects before the constructor so we can reference them directly in the asserts (reads cleaner and verifies that pass by reference is used)